src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -93,86 +93,62 @@ public String getDeploymentId () {
+                long maxOtpRunnerStartupTimeMillis = 5 * 60 * 1000;
+                if (taskHasTimedOut(statusCheckStartTime, maxOtpRunnerStartupTimeMillis)) {
+                    failJob("Job timed out while waiting for otp-runner to produce a status file!");
                     return;
                 }
-                long bundleDownloadSeconds = (System.currentTimeMillis() - bundleDownloadStartTime) / 1000;
-                LOG.info("Bundle downloaded in {} seconds!", bundleDownloadSeconds);
-                status.update("Building graph...", 30);
             }
-            // Once bundle is downloaded, we await the build (or download if graph already built) of the graph.
-            long graphCheckStartTime = System.currentTimeMillis();
-            String graphStatusUrl = String.join("/", ipUrl, GRAPH_STATUS_FILE);
-            while (!graphIsAvailable) {
+            // Wait for otp-runner to write a status that fulfills expectations of this job
+            statusCheckStartTime = System.currentTimeMillis();
+            boolean otpRunnerCompleted = false;
+            while (!otpRunnerCompleted && !status.error) {
                 // If the request is successful, the OTP instance has started.
-                waitAndCheckInstanceHealth("graph build/download check: " + graphStatusUrl);
-                graphIsAvailable = checkForSuccessfulRequest(graphStatusUrl);
-                // wait a maximum of 4 hours if building the graph, or 20 minutes if downloading a graph
-                long maxGraphBuildOrDownloadWaitTimeMillis = graphAlreadyBuilt ? 20 * 60 * 1000 : 4 * 60 * 60 * 1000;
-                if (taskHasTimedOut(graphCheckStartTime, maxGraphBuildOrDownloadWaitTimeMillis)) {
-                    failJob("Job timed out while waiting for graph build/download. If this was a graph building machine, it may have run out of memory.");
+                waitAndCheckInstanceHealth("otp-runner completion check: " + statusUrl);
+                otpRunnerCompleted = checkForOtpRunnerCompletion(statusUrl);
+                // wait a maximum of 5 hours if building a graph, or 1 hour if just starting a server
+                long maxOtpRunnerWaitTimeMillis = (graphAlreadyBuilt ? 5 : 1) * 60 * 60 * 1000;
+                if (taskHasTimedOut(statusCheckStartTime, maxOtpRunnerWaitTimeMillis)) {
+                    failJob("Job timed out while waiting for otp-runner to finish!");
                     return;
                 }
             }
-            // Check graph status and fail job if there was a failure.
-            String graphStatus = getUrlAsString(graphStatusUrl);
-            if (graphStatus == null || !graphStatus.contains("SUCCESS")) {
-                failJob("Failure encountered while building/downloading graph.");
+
+            // Check if an otp-runner status file check has already failed this job.
+            if (status.error) {
                 return;
```

Comment:
Should this return statement be moved up to just below the task time out check in the while statement? You can modify the while condition to remove the error check and then check for `status.error` within the while block. That seems better than checking for it twice (in while condition and down here).